### PR TITLE
Fix test suite on windows (wrt fake_ide and coq-makefile)

### PR DESCRIPTION
**Pull request by gares**
The PR includes a bunch of commit (by maxime, to be squashed) to add support for windows CI (appveyor) and to fix the test suite (mostly by me).

There is still 1 test failing (misc/deps/deps.out) where I think there i a `\r\n` versus `\n` issue.

  https://ci.appveyor.com/project/gares/coq

### CODE REVIEW

**Comment by silene** [comment]
What is the rationale for removing `set -e`?

**Comment by gares** [comment]
It is there, but in the `template/init.sh`  file so that all tests have it by default.

**Comment by Zimmi48** [comment]
The changes to `test-suite/success/Hints.v` look like you just remove from the test whatever does not work on Windows to make the test pass. If some stuff cannot be tested on Windows because it is not supported, could you rather make some tests platform-dependent?

**Comment by gares** [comment]
IIRC `Timeout` does not work on Windows, because `alarm` is Unix only.
So using it to bound a command is fine (it won't detect it diverges, but if it converges we are good).
Conversely, expecting it to make a command fail is not going to work.

I agree with @Zimmi48 that having a comment `(* portable-on: unix *)` that is used to skip some tests would be better, but if there is just one file using the feature I'm not sure it is worth it.

**Comment by maximedenes** [comment]
@Zimmi48 What is the point of testing that something takes at least 1 second?

**Comment by Zimmi48** [comment]
> What is the point of testing that something takes at least 1 second?

Indeed, this test is a bit weird anyways.

File: test-suite/Makefile [hunk 1]
```diff
@@ -170,6 +170,7 @@ summary.log:
 report: summary.log
 	$(HIDE)./save-logs.sh
 	$(HIDE)if [ -n "${TRAVIS}" ]; then find logs/ -name '*.log' -not -name 'summary.log' -exec 'bash' '-c' 'echo "travis_fold:start:coq.logs.$$(echo '{}' | sed s,/,.,g)"' ';' -exec cat '{}' ';' -exec 'bash' '-c' 'echo "travis_fold:end:coq.logs.$$(echo '{}' | sed s,/,.,g)"' ';'; fi
+	$(HIDE)if [ -n "${APPVEYOR}" ]; then find logs/ -name '*.log' -not -name 'summary.log' -exec 'bash' '-c' 'echo {}' ';' -exec cat '{}' ';' -exec 'bash' '-c' 'echo' ';'; fi
```
Comment (Zimmi48) [inline test-suite/Makefile]: Before this PR, when running the test-suite locally I get a warning about `TRAVIS` being undefined. Now I get two warnings:

```
Makefile:172: avertissement : variable « TRAVIS » indéfinie
Makefile:173: avertissement : variable « APPVEYOR » indéfinie
```

Can you do something about it?

**Comment by maximedenes** [comment]
I'll try to have a look at it, but can I merge this PR anyway, since the problem was already there? This is @SkySkimmer's code IIRC.

I have a lot of things to do so I'd rather schedule the resolution of thoses warnings with lower priority.

File: test-suite/Makefile (hunk 1 above)
Comment (gares) [inline test-suite/Makefile]: This is the Makefile of Coq that sets an option of `make` to emit warnings on undefined variables.
```
coq/Makefile:MAKE_OPTS := --warn-undefined-variable --no-builtin-rules
```

The I've no opinion about its usefulness

Comment (gares) [inline test-suite/Makefile]: FTR, the logs `test-suite/coq-makefile/*.log` contain many warnings because of that too

**Comment by SkySkimmer** [comment]
If you add these 2 lines in the Makefile the warning should go away
``` Makefile
TRAVIS?=
APPVEYOR?=
```

**Comment by gares** [comment]
https://ci.appveyor.com/project/gares/coq/build/appveyor~19
it worked!

**Comment by maximedenes** [comment]
Yep, but now let's wait for Travis to double check we didn't disrupt the test suite on Linux and OSX ;)

**Comment by maximedenes** [comment]
Sorry for noticing only now, but I'd prefer merging this in 8.7.

**Comment by JasonGross** [comment]
> IIRC Timeout does not work on Windows, because alarm is Unix only.

Timeout used to not work on Windows.  Now it works most of the time on
Windows (but is not very accurate).

**Comment by Zimmi48** [comment]
> Sorry for noticing only now, but I'd prefer merging this in 8.7.

That's your call of course, but this kind of PR is perfectly eligible to backporting, since it's only about fixing and infrastructure.

**Comment by gares** [comment]
After the rebase, something needs to be fixed wrt the timing scripts:
  https://ci.appveyor.com/project/gares/coq/build/appveyor~20

**Comment by gares** [comment]
@JasonGross does this error ring a bell?
```
make[1]: Leaving directory '/cygdrive/c/projects/coq/test-suite/coq-makefile/timing/after'
--- time-of-build-before.log.desired.processed	2017-07-20 07:15:29.535696500 -0700
+++ time-of-build-before.log.processed	2017-07-20 07:15:29.400697200 -0700
@@ -1,6 +1,12 @@
 COQDEP Fast.v
 COQDEP Slow.v
 COQC Slow.v
-Slow (real: ., user: ., sys: ., mem: ko)
+------
+real	m.s
+user	m.s
+sys	m.s
 COQC Fast.v
-Fast (real: ., user: ., sys: ., mem: ko)
+------
+real	m.s
+user	m.s
+sys	m.s
==========> FAILURE <==========
    coq-makefile/timing/run.sh...Error!
```

**Comment by Zimmi48** [comment]
It seems that this could be a transient failure. We should investigate this.

**Comment by JasonGross** [comment]
@gares did it fail to find `/usr/bin/time` and fall back on the `time` built-in that doesn't support formats?

**Comment by maximedenes** [comment]
@gares I resolved the `time` issue, but there are still two failures: `3441.v` and `4811.v` that I don't remember seeing before the rebase.

**Comment by maximedenes** [comment]
I suspect it may be a bug with `Timeout Time`.

**Comment by maximedenes** [comment]
@gares is it expected that `Time Timeout 1 command` can print:

```
Error: Timeout!
Finished failing transaction in 0.126 secs <0.125u,9.s> (failure)
```

?

**Comment by maximedenes** [comment]
Alleluia, I got everything to pass. Thanks for the hints, @gares 

https://ci.appveyor.com/project/maximedenes/coq-5elh3/build/appveyor~110

**Comment by Zimmi48** [comment]
Now backported in v8.7.

**Comment by Zimmi48** [comment]
I only tested Travis though, not appveyor.

**Pull request merged**